### DISCUSSION TITLE
How to define trainings and validation data?

### DISCUSSION THREAD

**Discussion by matthiaskoenig** [PEtab-dev/PEtab#605] [Q&A]
Hi all,

I often work with subsets of my experiments for training (i.e., parameter optimization) and an independent dataset for validation. It would be great if PEtab could easily support this distinction.

One possible approach could be adding a column, experimentType, in the experiments table, where values like training or test indicate whether a given experiment is used for parameter optimization or evaluation.

Currently, I would likely handle this by creating two separate PEtab problems—one for training and one for testing. I would then use the optimized parameters from the training set to evaluate the test dataset.

I’d be interested to hear how others approach this and whether there are alternative solutions within the current PEtab framework.

Looking forward to your input!

Best, Matthias

**Comment by dweindl** [comment]
PEtab doesn't deal with validation sets at all, so there is no clear PEtab-solution so far. I don't know how you are using the validation set exactly, but for something like early-stopping, I could imagine some PEtab extension. If it's just about simulation after parameter estimation is done, I'd consider it rather out of scope for PEtab.

Using two separate PEtab problems with shared files (where reasonable) sounds like the best way to handle that at the moment.

**Reply by matthiaskoenig** [reply]
Perfect. The more I thought about it the more sense it makes to keep validation outside of the actual problem. It is mainly about visualization and evaluation in the end.
